app/src/org/commcare/utils/TimeBoundOperation.java

```diff
@@ -0,0 +1,76 @@
+package org.commcare.utils;
+
+/**
+ * A time bound operation will run a synchronized operation in the background, with the confidence
```

Comment:
"in the background": isn't it blocking the current thread? If so, maybe say something like "on a new, blocking thread"